packages/react-events/src/dom/Tap.js

```diff
@@ -0,0 +1,664 @@
+
+  if (pointerType === 'touch' || (isValidButton && !isContextMenu)) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+/**
+ * Communicating gesture state back to components
+ */
+
+function dispatchStart(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const type = 'tap:start';
+  const onTapStart = props.onTapStart;
+  if (onTapStart != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapStart);
+  }
+}
+
+function dispatchChange(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const onTapChange = props.onTapChange;
+  if (onTapChange != null) {
+    const payload = state.isActive;
+    dispatchDiscreteEvent(context, payload, onTapChange);
+  }
+}
+
+function dispatchUpdate(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+) {
+  const type = 'tap:update';
+  const onTapUpdate = props.onTapUpdate;
+  if (onTapUpdate != null) {
+    const payload = {...state.gestureState, type};
+    dispatchUserBlockingEvent(context, payload, onTapUpdate);
+  }
+}
+
+function dispatchEnd(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const type = 'tap:end';
+  const onTapEnd = props.onTapEnd;
+  if (onTapEnd != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapEnd);
+  }
+}
+
+function dispatchCancel(
+  context: ReactDOMResponderContext,
+  props: TapProps,
+  state: TapState,
+): void {
+  const type = 'tap:cancel';
+  const onTapCancel = props.onTapCancel;
+  if (onTapCancel != null) {
+    const payload = {...state.gestureState, type};
+    dispatchDiscreteEvent(context, payload, onTapCancel);
+  }
+}
+
+/**
+ * Responder implementation
+ */
+
+const responderImpl = {
+  targetEventTypes,
+  getInitialState(): TapState {
+    return createInitialState();
+  },
+  onEvent(
+    event: ReactDOMResponderEvent,
+    context: ReactDOMResponderContext,
+    props: TapProps,
+    state: TapState,
+  ): void {
+    if (props.disabled) {
+      removeRootEventTypes(context, state);
+      if (state.isActive) {
+        dispatchCancel(context, props, state);
+        state.isActive = false;
+      }
+      return;
+    }
+
+    const nativeEvent: any = event.nativeEvent;
+    const eventTarget: Element = nativeEvent.target;
+    const eventType = event.type;
+
+    switch (eventType) {
+      // START
+      case 'pointerdown':
+      case 'mousedown':
+      case 'touchstart': {
+        if (hasPointerEvents) {
+          const pointerId = nativeEvent.pointerId;
+          state.activePointerId = pointerId;
+          // Make mouse and touch pointers consistent.
+          // Flow bug: https://github.com/facebook/flow/issues/8055
+          // $FlowExpectedError
+          eventTarget.releasePointerCapture(pointerId);
```

Comment:
Won't calling this throw on Safari/iOS as it's not on the prototype?

Reply:
It's wrapped in `if (hasPointerEvents)`

Reply:
We might still need to guard against lack of support, i.e. Firefox Mobile:

https://developer.mozilla.org/en-US/docs/Web/API/Element/releasePointerCapture

Reply:
Looks like Firefox for Android doesn't have PointerEvent enabled by default.